[docs][account-types]: Update the outdated information

amandeepmittal opened the pull request:
# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.
-->

Refs ENG-5758

# How

<!--
How did you build this feature or fix this bug and why?
-->

This PR:
- adds info on all different account types: Personal, Teams, Organization into their own sections
- clearly describes what roles are available
- adds info on how to create a new organization
- updates out the info around converting a Personal Account to Organization
- Adds a section for Teams account

---

🙏 Huge thanks to @kelley-sharp for pointing out the areas where the information was outdated and taking initiative on improving the documentation.

---

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->

Changes tested by running the docs locally.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

amandeepmittal committed:
[docs][account-types]: Update the outdated information

amandeepmittal committed:
Update account-types.md

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
 - Your EAS subscription will continue without interruption.
 - Your production apps will continue to operate without interruption.
 
+The Organization Account will adopt your old username. Converting a Personal Account into an Organization Account requires that you assign a new user as the owner of the Organization account. You will have the opportunity to sign up for a new Personal Account during the account conversion process and assign it or select an existing Personal Account (with a different username) to be the primary Ow ...
```
```suggestion
The Organization Account will adopt your old username. Converting a Personal Account into an Organization Account requires that you designate a new user as the owner of the Organization account. You will have the opportunity to designate an existing user or sign up as a new user during the account conversion process.
```

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
+
+<ImageSpotlight alt="Example demonstrating the Step 3 of the conversion process." src="/static/images/accounts/converting-personal-account.jpg" style={{maxWidth: 720}} />
+
+After completing the conversion process, you can no longer access the Personal Account you converted. To continue using Expo services, you will log into [expo.dev](https://expo.dev/) with the Personal Account you selected or created to manage the Organization Account. Then, select the organization from the top left dropdown to access the organization.
```
```suggestion
After completing the conversion process, you can no longer log in as the old Personal Account's user. To continue using Expo services, you will log into [expo.dev](https://expo.dev/) with the user you selected or created to manage the Organization Account. Then, select the organization from the top left dropdown to access the organization.
```

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
+
+After completing the conversion process, you can no longer access the Personal Account you converted. To continue using Expo services, you will log into [expo.dev](https://expo.dev/) with the Personal Account you selected or created to manage the Organization Account. Then, select the organization from the top left dropdown to access the organization.
+
+> **Note**: Before account conversion, make sure that all of your projects are on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for notifications to continue to work on those projects.
```
```suggestion
```

(this is only true for account renaming and project transfer)

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
 ### Renaming an Account
 
-Accounts may be renamed a limited number of times. Simply visit [the account settings page](https://expo.dev/accounts/[account]/settings) and follow the prompts under **Rename Account**.
+Accounts can be renamed a limited number of times. Members in an Organization with Owners or Admin roles can rename the Account. Visit [the account settings](https://expo.dev/accounts/[account]/settings) and follow the steps under [Rename Account](https://expo.dev/settings#rename-account).
```
```suggestion
Accounts can be renamed a limited number of times. Only Owners may rename accounts. Visit [the account settings](https://expo.dev/accounts/[account]/settings) and follow the steps under [Rename Account](https://expo.dev/settings#rename-account).
```

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
+#### Caveats
 
-#### Some caveats
+Before renaming an account, make sure that your project is on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for OTA updates to continue to work on that project.
```
```suggestion
After renaming an account, new publishes to projects within the account must be on Expo SDK 43 or above.
```

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
 
-- The person performing the transfer must have "Owner" role on both the source and destination accounts.
-- New publishes for renamed project must be on SDK 43 or higher.
+Before transferring an account, make sure that your project is on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for OTA updates to continue to work on that project.
```
```suggestion
Before transferring a project, make sure that your it is on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for OTA updates to continue to work on that project.
```

wschurman reviewed (changes requested):
Higher-level points of discussion:
- This seems like a good time to differentiate between users and accounts in our docs.
- We may want to just bury/deprecate/etc team account docs. I see the note, but it should probably be the last section on the page and not mentioned in intro. I'm not sure of the status of this effort though.

kelley-sharp commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
 
+import ImageSpotlight from '~/components/plugins/ImageSpotlight';
+
+There are three types of Expo accounts; Personal, Team, and Organization. The type you choose depends on whether you work alone on projects or in a group. However, you must start with at least one Personal Account to use Expo.
```
I advised that we still include Team Accounts in the documentation for now since [this discussion](https://exponent-internal.slack.com/archives/CTR08334G/p1659653015542189) is still pending

amandeepmittal replied:
We'll have to conclude internally as to whether include Team related info in docs or not. As Kelley mentioned, I followed her suggestion.

My suggestion, is to keep the info for now since its still a feature available but remove it as soon as Team accounts functionality are disabled.

kelley-sharp commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
-Every user of Expo has their own Personal Account. With this account, we can safely identify you as the developer or owner of an app. You can use Personal Accounts for all features Expo has to offer. In cases where you take an action that doesn't specify which account it should belong to, it will default to your Personal Account
+When you sign up for an account with Expo, you are creating a Personal Account. All Expo users have their own Personal Account(s). Expo uses it to identify you as an individual safely.
+
+A Personal Account is the default account type. For example, your Personal Account username is used when you use the [contact form](https://expo.dev/contact) to reach out for support when you are logged in.
```
I think it's sufficient to remove this line.

amandeepmittal commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
+
+When you [sign up for an account](https://expo.dev/signup) with Expo, you are creating a Personal Account. If you want multiple members to have access to project(s), you can create an Organization Account from within your Personal Account and transfer projects individually to the Organization account, if required. Another option is to convert your Personal Account directly to an Organization Acco ...
+
+Before Organization accounts, the only way to work in a group was to add members directly to a Personal Account This is called a Team Account. Only the original member can have an Owner role in a Team Account. An Organization Account can have multiple members with the Owner role. You can choose whichever account type that makes the most sense for your group. Both group account types allow the fol ...
```
I'll move the Team Account at the end of the page (please see my comment: https://github.com/expo/expo/pull/18552/files#r942451815) but I am also open to remove it completely from the page or updating the section title as:

```markdown
## Team (legacy).
```

There is a note in the Team account section that directs the user to Organization. here is the note:

```markdown
> **Note**: For all new accounts where you want to share access, we recommend you [create a new Organization](#organizations) Account. It allows more flexibility in role assignments and allows you to keep some projects private to you. However, a Team Account is still a viable way to give other users access to all your projects, just keep in mind that no projects in your account can be private to you if you make a Team Account unless you remove all members.
```

amandeepmittal commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
+> **Note**: For all new accounts where you want to share access, we recommend you [create a new Organization](#organizations) Account. It allows more flexibility in role assignments and allows you to keep some projects private to you. However, a Team Account is still a viable way to give other users access to all your projects, just keep in mind that no projects in your account can be private to  ...
 
-> This account is just for you, **never share access to this account for any reason.**
+To create a Team Account, simply [invite Team members](/accounts/working-together/#adding-members) to a Personal Account. An email will be sent that prompts them to create a Personal Expo Account if they do not already have one. In a Team Account, the team's creator has the Owner role only.
```
The changes you suggested sound confusing to me. How about:

```suggestion
To create a Team Account, simply [invite Team members](/accounts/working-together/#adding-members) to a Personal Account. An email will be sent that prompts them to create a Personal Expo Account if they do not already have one. Only the person (for whom a Personal Account was automatically created) is the Owner of that Personal Account and, therefore, the Team Account.
```

amandeepmittal commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
 ## Organizations
 
-If you are creating a project that may need to outlive your involvement with it, for instance a professional project for a company you work for, we recommend you create a new Organization for that project.
+An Organization Account is useful for creating a project that may outlive your involvement. If you are working on a shared project with other members of a company or a group of developers, we recommend you [create an Organization](expo.dev/create-organization) Account.
```
Thanks for the suggestion! I'll update this section separately, I think I've got the idea of what is required here.

amandeepmittal commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
-- Sharing access with others
-- Isolating expenses for work projects that you need to submit expenses for
-- Structuring projects for different contexts (e.g. work for different clients)
+To create a new Organization, log in to your Personal Account and [open the link](https://expo.dev/create-organization) to create a new organization.
```
Didn't know about the difference here. Thanks!

amandeepmittal commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
-| **Transfer control of all projects to another user**                |                   | ✅           |
-| **Programmatic access with limited privileges**                     |                   | ✅           |
-| **Designate multiple users who have complete control of a project** |                   | ✅           |
+Alternatively, if you are logged in to your Personal Account, you can create a new Organization from the dashboard:
```
In that case, I'll remove the previous changes and stick to this version.

amandeepmittal committed:
Apply suggestions from code review

Co-authored-by: Will Schurman <wschurman@gmail.com>

amandeepmittal commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,115 @@
+- Assigning member permissions (only one Owner in a Team Account)
+- Sharing an [EAS Subscription](https://docs.expo.dev/eas/)
+
+Original or added Owners can invite and assign members of their group a role. This role determines what actions they can or can not perform. Four types of roles are available:
```
In the latest PR, I'm removing this part and merging the bullet points above in the organization section. After removing the team (and deprecating it at the bottom of the page), I do not see a point in explaining "group" accounts here separately.

amandeepmittal committed:
Apply code review changes and add a new section on inviting members

amandeepmittal committed:
Fix

amandeepmittal commented:
In the latest commit:

- I've trimmed down the introduction to not include mention of "Teams" any more 
- Moved the "Teams" section at the bottom of the page (making the doc future-proof to easily extract any info related to "Teams" whenever there is a conclusion on it)
- Since there is no "Teams" in the intro anymore, move all info related to "group accounts" and role privileges into "Organizations" and create a new section on "Inviting members".
- Updated "Organizations" section intro as Will's suggestions.

amandeepmittal committed:
Fix note placement

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,112 @@
+
+<ImageSpotlight alt="Example demonstrating the Step 3 of the conversion process." src="/static/images/accounts/converting-personal-account.jpg" style={{maxWidth: 720}} />
+
+After completing the conversion process, you can no longer log in as the old Personal Account's user. To continue using Expo services, you will log into [expo.dev](https://expo.dev/) with the user you selected or created to manage the Organization Account. Then, select the organization from the top left dropdown to access the organization.
```
```suggestion
After completing the conversion process, you will no longer be able to log in as the your old user. To continue using Expo services, log into [expo.dev](https://expo.dev/) with the user you selected or created during the conversion process. Then, select the organization from the top left dropdown to access the organization.
```

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,112 @@
+
+### Inviting members
+
+Since the Organization is a shared container with your team, you can invite other members to it. To invite a new member:
```
```suggestion
Other Expo users can be invited to join your Organization. To invite a new member:
```

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,112 @@
+- From the Organization's dashboard, select Organization settings from the sidebar. This will open all the settings available for the organization.
+- In Account settings, select Members from the sidebar.
+- Under Members, click the button "Invite Member". This will open a form to invite a member to the organization.
+- In the form, enter the email of the member you want to invite and select their role. For more information, see [role privileges](/accounts/working-together/#managing-access).
```
```suggestion
- In the form, enter the email of the user you want to invite and select the role they should have upon joining the organization. For more information, see [role privileges](/accounts/working-together/#managing-access).
```

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,112 @@
+
+When inviting a new member, keep in mind:
+
+- Only a member with an Admin or an Owner role can invite others.
```
```suggestion
- Only members with an Admin or an Owner role can invite others.
```

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,112 @@
+When inviting a new member, keep in mind:
+
+- Only a member with an Admin or an Owner role can invite others.
+- Member with an Owner role can give members/invitees any role.
```
```suggestion
- Members with an Owner role can grant members and invitees any role.
```

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,112 @@
+
+- Only a member with an Admin or an Owner role can invite others.
+- Member with an Owner role can give members/invitees any role.
+- Member with an Admin role can only give members/invitees up to and including Admin role.
```
```suggestion
- Members with an Admin role can only give members and invitees up to and including Admin role (every role but Owner).
```

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,112 @@
 ### Renaming an Account
 
-Accounts may be renamed a limited number of times. Simply visit [the account settings page](https://expo.dev/accounts/[account]/settings) and follow the prompts under **Rename Account**.
+Accounts can be renamed a limited number of times. Only Owners can rename accounts. Visit [the account settings](https://expo.dev/accounts/[account]/settings) and follow the steps under [Rename Account](https://expo.dev/settings#rename-account).
```
```suggestion
Accounts can be renamed a limited number of times. Only Owners can rename accounts. To rename an account, visit [the account settings](https://expo.dev/accounts/[account]/settings) and follow the steps under [Rename Account](https://expo.dev/accounts/[account]/settings#rename-account).
```

wschurman commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,112 @@
+#### Caveats
 
-- New publishes for projects belonging to renamed accounts must be on SDK 43 or higher.
+Before transferring a project, make sure that your it is on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for OTA updates to continue to work on that project.
```
```suggestion
Before transferring a project, make sure that it is on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for OTA updates to continue to work on that project.
```

amandeepmittal commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,113 @@
+
+When you [sign up for an account](https://expo.dev/signup) with Expo, a Personal Account is automatically created for you. If you want multiple members to have access to project(s), you can create an Organization Account.
+
+Before Organization accounts, the only way to work in a group was to add members directly to a Personal Account. This was called a Team Account. Only the original member can have an Owner role in a Team Account. An Organization Account can have multiple members with the Owner role. You can choose whichever account type that makes the most sense for your group. Both group account types allow the f ...
```
> move the history of team accounts and other team account info to the team account section

I've removed the history of team accounts from the introduction. I do not think its relevant. Let me know if you think it should be there.

>move the organization account info to the organization account section

Re-wrote Organization section introduction which includes the new info in detail.

>as for the role stuff, I think it's fine to just link to the working-together page and not get into too much detail here. We may need to update that page though, idk.

Yeah, I agree on, that's what I've done. I did open an issue in Linear (https://linear.app/expo/issue/ENG-5975/verify-and-update-outdated-info-on-roles-privileges) to update that page and have some questions. When you have time, can you please take a look? 🤗

amandeepmittal commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,113 @@
+
+Each role has different privileges. Learn more about [role privileges](/accounts/working-together/#managing-access).
+
+> **Note**: You can switch between accounts or create a new Organization Account from the dropdown in the top left corner. If you want to convert your Personal Account to an Organization Account, select the avatar from the top right corner of the dashboard and then select User Settings.
```
The section `Converting Personal Accounts into Organizations` already includes these instructions so removed the note from into.

amandeepmittal commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,113 @@
+When you sign up a Personal Account is automatically created for you. This account is a good place to work on your personal projects.
+
+
+> **Note**: You must not share authentication credentials for your Personal Account with anyone for any reason. Members can log in to their Personal Accounts to access any Team or Organization they are part of. Access to these accounts is available from the dropdown in the top left corner of the dashboard on [expo.dev](https://expo.dev).
```
Removed the second part and I think in each Team and Organization section talks about accessing them in detail in their own section. So removed it altogether.

amandeepmittal commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,113 @@
+
+## Teams
+
+> **Note**: For all new accounts where you want to share access, we recommend you [create a new Organization](#organizations) Account. It allows more flexibility in role assignments and allows you to keep some projects private to you. However, a Team Account is still a viable way to give other users access to all your projects, just keep in mind that granting someone access to your account grants ...
```
Updated the first part of the note as per your suggestion (that is to remove "and allows you to keep some projects private to you").

Replaced the second part " However, a Team Account is still a viable way to give other users access to all your personal projects, just keep in mind that granting someone access to your Personal account grants them access to all of your personal projects." with "Creating a Team Account is deprecated."

amandeepmittal committed:
Apply changes as per code review from Will

amandeepmittal commented:
@wschurman, applied all suggestions in one commit (latest commit). Thanks once again for the in-depth review 👍

kelley-sharp commented on docs/pages/accounts/account-types.md:
```diff
@@ -2,69 +2,110 @@
 
-- The person performing the transfer must have "Owner" role on both the source and destination accounts.
-- New publishes for renamed project must be on SDK 43 or higher.
+> **Note**: For all new accounts where you want to share access, we recommend you to [create a new Organization](#organizations) Account. It allows more flexibility in role assignments Creating a Team Account is deprecated.
```
missing period

amandeepmittal committed:
Update docs/pages/accounts/account-types.md

Co-authored-by: Kelley Sharp <kelley@expo.dev>
